pkgs/development/python-modules/dgl/default.nix

```diff
@@ -0,0 +1,261 @@
+    "test_sparse_opt"
+  ];
+
+  commonDisabledTests = [
+    "test_actor"
+    "test_add_node_property_split"
+    "test_add_nodepred_split"
+    "test_adj[idtype0]"
+    "test_adj[idtype1]"
+    "test_as_graphpred"
+    "test_as_graphpred_ogb"
+    "test_as_graphpred_reprocess"
+    "test_as_linkpred"
+    "test_as_linkpred_ogb"
+    "test_as_nodepred1"
+    "test_as_nodepred2"
+    "test_as_nodepred_ogb"
+    "test_chameleon"
+    "test_citation_graph"
+    "test_cluster"
+    "test_cornell"
+    "test_explain_syn"
+    "test_fakenews"
+    "test_flickr"
+    "test_fraud"
+    "test_gather_mm_idx_b[dtype1-0.02-256]"
+    "test_gin"
+    "test_global_uniform_negative_sampling[int32]"
+    "test_global_uniform_negative_sampling[int64]"
+    "test_gnn_benchmark"
+    "test_pattern"
+    "test_reddit"
+    "test_squirrel"
+    "test_texas"
+    "test_tudataset_regression"
+    "test_wiki_cs"
+    "test_wisconsin"
+    "test_zinc"
+  ];
+
+  distributedDisabledTests = [
+    "test_dataloader[edge-0-3]"
+    "test_dataloader[edge-4-3]"
+    "test_dataloader[node-0-3]"
+    "test_dataloader[node-4-3]"
+    "test_dist_dataloader[1-False-0-3]"
+    "test_dist_dataloader[1-False-4-3]"
+    "test_dist_dataloader[1-True-0-3]"
+    "test_dist_dataloader[1-True-4-3]"
+    "test_dist_emb_server_client"
+    "test_dist_optim_server_client"
+    "test_kv_multi_role"
+    "test_kv_store"
+    "test_multi_client[socket]"
+    "test_multi_client[tensorpipe]"
+    "test_multi_client_connect[socket]"
+    "test_multi_client_connect[tensorpipe]"
+    "test_multi_thread_rpc[socket]"
+    "test_multi_thread_rpc[tensorpipe]"
+    "test_multiple_dist_dataloaders[edge-0-1]"
+    "test_multiple_dist_dataloaders[edge-0-4]"
+    "test_multiple_dist_dataloaders[edge-1-1]"
+    "test_multiple_dist_dataloaders[edge-1-4]"
+    "test_multiple_dist_dataloaders[edge-4-1]"
+    "test_multiple_dist_dataloaders[edge-4-4]"
+    "test_multiple_dist_dataloaders[node-0-1]"
+    "test_multiple_dist_dataloaders[node-0-4]"
+    "test_multiple_dist_dataloaders[node-1-1]"
+    "test_multiple_dist_dataloaders[node-1-4]"
+    "test_multiple_dist_dataloaders[node-4-1]"
+    "test_multiple_dist_dataloaders[node-4-4]"
+    "test_neg_dataloader[0-3]"
+    "test_neg_dataloader[4-3]"
+    "test_rpc[tensorpipe]"
+    "test_rpc_find_edges_shuffle[1]"
+    "test_rpc_find_edges_shuffle[2]"
+    "test_rpc_get_degree_shuffle[1]"
+    "test_rpc_get_degree_shuffle[2]"
+    "test_rpc_in_subgraph"
+    "test_rpc_sampling_shuffle[1]"
+    "test_rpc_sampling_shuffle[2]"
+    "test_rpc_timeout[socket]"
+    "test_rpc_timeout[tensorpipe]"
+    "test_server_client"
+    "test_standalone"
+    "test_standalone_etype_sampling"
+    "test_standalone_sampling"
+  ];
+
+  disabledTestString = testsList: "not (${lib.strings.concatStringsSep " or " testsList})";
+in
+
+buildPythonPackage rec {
+  pname = "dgl";
+  version = "1.1.3";
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner = "dmlc";
+    repo = pname;
```

Comment:
Please don't use this.

`repo = pname;` is nice for DRY but creates a binding that goes too far.

See further information about this here: https://github.com/nix-community/nixpkgs-lint/issues/21